src/utils/sample/discrete_gauss.rs

```diff
@@ -256,3 +350,97 @@ mod test_gaussian_function {
         let _ = gaussian_function(&sample, &center, &gaussian_parameter);
     }
 }
+
+#[cfg(test)]
+mod test_sample_d {
+    use crate::utils::sample::discrete_gauss::sample_d;
+    use crate::{
+        integer::{MatZ, Z},
+        rational::{MatQ, Q},
+    };
+    use std::str::FromStr;
+
```

Comment:
I would like some sort of test, which ensures that the sampled vector, lies in the lattice.
I am certain it does, as you only add a basis vector times a scalar in Z every time, but a test woould be nice none-the-less.